clang/test/SemaSYCL/attr-syclglobalvar.cpp

```diff
@@ -0,0 +1,21 @@
+// RUN: %clang_cc1 -fsycl-is-device -verify -fsyntax-only %s
+
+#include "Inputs/sycl.hpp"
+
+__attribute__((sycl_global_var)) int GlobalWithAttribute;
+
+int GlobalNoAttribute;
+
+// expected-error@+1 {{attribute takes no arguments}}
+__attribute__((sycl_global_var(42))) int GlobalWithAttributeArg;
+
+// expected-warning@+1 {{attribute only applies to global variables}}
+__attribute__((sycl_global_var)) void F() {
+    // expected-warning@+1 {{attribute only applies to global variables}}
+    __attribute__((sycl_global_var)) int Local;
+
+    cl::sycl::kernel_single_task<class kernel_name>([=] () {
+        (void)GlobalWithAttribute;
+        (void)GlobalNoAttribute; // expected-error {{SYCL kernel cannot use a non-const global variable}} expected-note@Inputs/sycl.hpp:* {{called by}}
+    });
+}
```

Comment:
Please add a newline to the end of the file, as well as the more complex tests from above. Some additional tests as well:
```
namespace NS {
__attribute__((sycl_global_var)) int i; // This should definitely be a global variable that works
}

__attribute__((sycl_global_var)) extern int i; // This should work.

void func() {
  __attribute__((sycl_global_var)) extern int j; // Should this work?
  __attribute__((sycl_global_var)) static int k; // Definitely should not work
}

__attribute__((sycl_global_var)) thread_local int l; // Definitely a global variable, but should this work?
```

Reply:
Latest commit fixed newline and expanded test cases and docs.